utils/RuleReader.py

```diff
@@ -0,0 +1,69 @@
+import os
+import json
+
+class RuleReader:
+    def __init__(self, serviceFolderPath, service=None):
+        self.serviceFolderPath = serviceFolderPath
+        self.service = service
+        # self.reporterPathDict = {}
+        
+        # self.setReporterPathDict()
+        return
+    
+    def getReporterPathList(self):
+        service = self.service
+        folderPathList = []
+        reporterPathList = []
+        if service is None:
+            with os.scandir(self.serviceFolderPath) as dir:
+                directories = list(dir)
+            directories.sort(key=lambda x: x.name)
+            
+            for item in directories:
+                if item.is_dir():
+                    folderPath = self.serviceFolderPath + '/' + item.name + '/'
+                    folderPathList.append(folderPath)
+        else:
+            folderPath = self.serviceFolderPath + '/' + service + '/'
+            if os.path.isdir(folderPath):
+                folderPathList.append(folderPath)
+            else:
+                print("Service folder " + folderPath + " not found")
+            
+        for path in folderPathList:
+            folderItems = os.scandir(path)
+            for item in folderItems:
+                if item.is_file() and item.name.endswith('reporter.json'):
+                    reporterPathList.append(path + item.name)
+            
+        return reporterPathList
+    
+    def getRulesFromReporter(self):
+        service = self.service
+        pathList = self.getReporterPathList()
+        rules = {}
+        for path in pathList:
+            reporterFile = open(path, 'r')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *reporterFile*. The resource is allocated by call *builtins.open*. The resource is referenced at lines 47, 48. The resource is closed at line 48. There are other execution paths that do not contain closure statements, for example, when *TextIOBase.read()* throws an exception. To prevent this resource leak, close *reporterFile* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).